### PR TITLE
Fix the pulsar sql can not query the last message issue

### LINKED ISSUE #7811: Pulsar SQL can't query the InactiveTopics and un-managedLedger
## **Describe the bug**
Pulsar SQL can't query the InactiveTopics and un-managedLedger 

### Pulsar SQL  [PulsarSplitManager.getSplitsForTopic()](https://github.com/apache/pulsar/blob/master/pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java#L261) split with this code:
``readOnlyCursor = managedLedgerFactory.openReadOnlyCursor(
                    topicNamePersistenceEncoding,
                    PositionImpl.earliest, new ManagedLedgerConfig());``

this will producing two unexpected result for me :

### 1) when producer expired with brokerDeleteInactiveTopicsMaxInactiveDurationSeconds ,the topic will log this on **broker** :

``08:31:43.073 [bookkeeper-..] INFO  org.apache.pulsar.broker.service.persistent.PersistentTopic - [persistent://aa/bb/cc-partition-0] Topic deleted successfully due to inactivity``

 that will raise exception on  **presto** :

``org.apache.bookkeeper.mledger.ManagedLedgerException$ManagedLedgerNotFoundException: org.apache.bookkeeper.mledger.ManagedLedgerException$MetadataNotFoundException: Managed ledger not found``
### 2) adjust brokerDeleteInactiveTopicsMaxInactiveDurationSeconds to infinite :

[PulsarSplitManager.getSplitsForTopic()](https://github.com/apache/pulsar/blob/master/pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java#L261) only return the managedLedger (look like the Ledger with writing status, only one Ledger for one partiton  ) , that case lost read the history data。

### **Expected behavior**
Pulsar SQL can read history or Inactive Topic data

_is there any interface that  can read all history Ledger data ?_

### **Additional context**
TAG : `managed-ledger` `Pulsar SQL`

### PR DESCRIPTION
**Pull request by zymap**
Fixes #7811

### Motivation

Currently, the bookkeeper client gets the LAC is the piggyback lac which is carried by the next message. This causes the pulsar SQL connector will always get the `message number -1` messages. If we want to query all messages, we need to use the explicit LAC to get the total number of entries. 
BookKeeper 4.12.0 makes us can use the `useV2Protocol` to enable using v3 protocol which will get the explicit LAC.

### Modifications

- Add the properties for the pulsar SQL.
- Update the integration test to query the last message.

### CODE REVIEW

**Review by jiazhai — APPROVED** [review]

File: conf/broker.conf [hunk 1]
```diff
@@ -774,6 +774,9 @@ bookkeeperDiskWeightBasedPlacementEnabled=false
 # A value of '0' disables sending any explicit LACs. Default is 0.
 bookkeeperExplicitLacIntervalInMills=0
 
+  # Use older Bookkeeper wire protocol with bookie
+bookkeeperUseV2WireProtocol=true
```
Comment (codelipenghui) [inline conf/broker.conf:778]: I think we already have this config in the broker.conf https://github.com/apache/pulsar/blob/master/conf/broker.conf#L680

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java [hunk 2]
```diff
@@ -72,6 +72,8 @@
     private int bookkeeperThrottleValue = 0;
     private int bookkeeperNumIOThreads = 2 * Runtime.getRuntime().availableProcessors();
     private int bookkeeperNumWorkerThreads = Runtime.getRuntime().availableProcessors();
+    private boolean useV2Protocol = true;
```
Comment (codelipenghui) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java:75]:
```suggestion
    private boolean bookkeeperUseV2Protocol = true;
```

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java [hunk 3]
```diff
@@ -72,6 +72,8 @@
...
+    private int explicitInterval = 0;
```
Comment (codelipenghui) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java:76]:
```suggestion
    private int bookkeeperExplicitInterval = 0;
```

**Review by codelipenghui — CHANGES_REQUESTED** [review]

**Comment by zymap** [comment]
@codelipenghui I addressed your comments.PTAL again. Thanks.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**